### PR TITLE
Update to Django 1.7

### PR DESCRIPTION
**Pull request by Hwesta**
Update to use Django 1.7, including fixes for backwards compatibliity and other versions bumps.

### CODE REVIEW

**Comment by mistydemeo** [comment]
:+1:

**Comment by Hwesta** [comment]
Updated to handle converting from South to Django migrations.  Ready to be looked at again. Should be tested upgrading packages from 0.3, 0.4 and 0.5.

File: debian/postinst [hunk 1]
```diff
@@ -28,7 +30,16 @@ rsync -a /var/archivematica/storage-service/templates /usr/lib/archivematica/sto
...
 
 echo "configuring django database and static files"
-/usr/share/python/archivematica-storage-service/bin/python manage.py syncdb
+find /usr/lib/archivematica/storage-service/ -name '*.pyc'  -delete
+# 0.3.0 will automatically fake the first migration
+if [[ $2 == '0.6.'* || $2 == '0.5.'* || $2 == '0.4.'* ]]; then
+    echo 'Faking the 0.4 migration'
+    /usr/share/python/archivematica-storage-service/bin/python manage.py migrate locations 0002_v0_4 --fake
+fi
+if [[ $2 == '0.6.'* || $2 == '0.5.'* ]]; then
```
Comment (Hwesta) [inline debian/postinst]: Check with Justin that these actually match the versions postinst will be called with.

File: requirements/base.txt [hunk 2]
```diff
@@ -1,10 +1,9 @@
 # Base requirements - for all installations
-Django==1.5.4
+Django>=1.7,<1.8
```
Comment (mistydemeo) [inline requirements/base.txt:2]: 🚀

File: storage_service/storage_service/settings/test.py [hunk 3]
```diff
@@ -1,10 +1,7 @@
 from base import *
 
 ########## TEST SETTINGS
-TEST_RUNNER = 'discover_runner.DiscoverRunner'
-TEST_DISCOVER_TOP_LEVEL = SITE_ROOT
-TEST_DISCOVER_ROOT = SITE_ROOT
-TEST_DISCOVER_PATTERN = "test_*.py"
+# TEST_RUNNER = 'django.test.runner.DiscoverRunner'
```
Comment (mistydemeo) [inline storage_service/storage_service/settings/test.py]: Commented out?

**Comment by mistydemeo** [comment]
Haven't tested but looks good :+1:

**Comment by Hwesta** [comment]
Added south migration for 0.7

**Comment by mistydemeo** [comment]
:+1:

File: storage_service/storage_service/settings/test.py (hunk 3 above)
Comment (Hwesta) [inline storage_service/storage_service/settings/test.py]: Looks like it was moved to base.py and I forgot to delete it here. Fixed.

**Comment by mistydemeo** [comment]
:shipit:

**Pull request merged**